README.md

```diff
@@ -1372,6 +1374,97 @@ module.exports = {
 };
 ```
 
+##### `getJSON`
+
+Type:
+
+```ts
+type getJSON = (resourcePath: string, json: object) => any;
+```
+
+Default: `undefined`
+
+Enables a callback to output the CSS modules mapping JSON. The callback is invoked with two arguments:
+
+- `resourcePath`: the absolutely path of the original resource, e.g., /foo/bar/baz.css
+- `json`: the CSS modules map object, e.g.,
+
+```
+/* baz.css */
+
+.a {
+  background-color: aliceblue;
+}
+
+.b {
+  background-color: burlywood;
+}
+```
+
+`json` will be something like the following (depending on your other `modules` settings):
+
+```
+{
+  "a": "a__uRkh1",
+  "b": "b__pjFcy"
+}
+```
+
+**webpack.config.js**
+
+```js
+// supports a synchronous callback
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.css$/i,
+        loader: "css-loader",
+        options: {
+          modules: {
+            getJSON: (resourcePath, json) => {
+              // synchronously write a .json mapping file in the same directory as the resource
+              const outputPath = path.resolve(
+                path.dirname(resourcePath),
+                `${path.basename(resourcePath)}.json`
+              );
+
+              const fs = require("fs");
+              fs.writeFileSync(outputPath, JSON.stringify(json));
```

Comment:
Let's improve the example and show how to write it in the one file, because it is a prefered way instead a lot of fs read calls

Reply:
With structure like:
```
{
  "resource-path.css": {
    "foo": "abcdef0987654321"
  }
}
```

Reply:
Note - additional example and keep this

Reply:
Ooh, that definitely makes sense to me, but I'm not sure how to determine when we're "done" to write out the aggregated file.  Like it's trivial to keep track of a Map/object in memory and add to it as `getJSON` is called, but it wouldn't know when to write out since each invocation of the loader function occurs in the context of an individual resource.  Is there something like a lifecycle hook or a callback exposed to Webpack loaders like with plugins?

(Apologies if this is a dumb question--this is the first time I've worked on a Webpack loader.  Thanks for all your guidance in any case!)